lib/compiler/passes/generate-js.js

```diff
@@ -707,16 +707,19 @@ function generateJS(ast, options) {
       "      }",
       "    }",
       "    var s = this.location.start;",
-      "    var loc = this.location.source + \":\" + s.line + \":\" + s.column;",
+      "    var offset_s = (typeof this.location.source.offset === \"function\")",
+      "      ? this.location.source.offset(s)",
+      "      : s;",
+      "    var loc = this.location.source + \":\" + offset_s.line + \":\" + offset_s.column;",
       "    if (src) {",
       "      var e = this.location.end;",
-      "      var filler = peg$padEnd(\"\", s.line.toString().length, ' ');",
+      "      var filler = peg$padEnd(\"\", offset_s.line.toString().length, ' ');",
       "      var line = src[s.line - 1];",
       "      var last = s.line === e.line ? e.column : line.length + 1;",
       "      var hatLen = (last - s.column) || 1;",
       "      str += \"\\n --> \" + loc + \"\\n\"",
       "          + filler + \" |\\n\"",
-      "          + s.line + \" | \" + line + \"\\n\"",
+      "          + offset_s.line + \" | \" + line + \"\\n\"",
```

Comment:
Would it be better to retrieve newline characters from NodeJS environment itself?

Reply:
We don't have access to the node environment here.  This is generated es5 JS that needs to be able to run in the browser also.  We could add an optional second parameter that is the line endings to use, then pass in node's `os.EOL` when calling this code from the CLI.  However, since this is an existing problem not related to this change, I'll start a new bug for us to discuss this.

Reply:
See #320

Reply:
I see. In my opinion, it should be possible to live the implementation as is at this point :|